scripts/buildhost/start_ec2_instance.sh

```diff
@@ -8,63 +8,84 @@
 #
 # Exit Code: returns 0 if instance started successfully, non-zero otherwise
 #
-# Examples: scripts/buildhost/start_ec2_instance.sh <AWS_REGION> <AWS_AMI>
+# Examples: scripts/buildhost/start_ec2_instance.sh <AWS_REGION> <AWS_AMI> <AWS_INSTANCE_TYPE>
 #
 # Upon successfull execution, the following files would be created:
 # sgid - contain the security group identifier
 # key.pem - contains the certificate required to log on to the server
 # instance - contains the address of the created instance
 #
 
+set -eo pipefail
+
 AWS_REGION=$1
 AWS_AMI=$2
 AWS_INSTANCE_TYPE=$3
 INSTANCE_NUMBER=$RANDOM
 KEY_NAME="BuilderInstanceKey_${INSTANCE_NUMBER}"
 SECURITY_GROUP_NAME="BuilderMachineSSH_${INSTANCE_NUMBER}"
 
-SGID=$(aws ec2 create-security-group --group-name ${SECURITY_GROUP_NAME} --description "Security Group for ephermal build machine to allow port 22" --region ${AWS_REGION} | jq -r '.GroupId')
-if [ "$?" != "0" ]; then
+function security_group_exists {
+    if aws ec2 describe-security-groups --group-name "${SECURITY_GROUP_NAME}" --region "$AWS_REGION" > /dev/null 2>&1; then
+        echo "WARNING: security group ${SECURITY_GROUP_NAME} exists"
+	    return 0
+    fi
+    
+    return 1
+}
+
+function key_pair_exists {
+    if aws ec2 describe-key-pairs --key-names "${KEY_NAME}" --region "$AWS_REGION" > /dev/null 2>&1; then
+        echo "WARNING: key pair ${KEY_NAME} exists"
+	    return 0
+    fi
+        return 1
+}
```

Comment:
Nitpick:  there is inconsistent whitespace in the two functions above.

Reply:
Tabs are evil. Removed.